### PR TITLE
Adding the custom handwired 5x5 macropad.

### PR DESCRIPTION
**Pull request by m40-dev**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/m40/5x5_macropad/readme.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# m40
+
+![m40](https://imgur.com/bWXH3FV)
```
Comment (drashna) [inline keyboards/handwired/m40/5x5_macropad/readme.md:3]:
```suggestion
![m40](https://i.imgur.com/bWXH3FVl.jpeg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/m40/5x5_macropad/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/handwired/m40/5x5_macropad/rules.mk:18]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/handwired/m40/5x5_macropad/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# m40
+
+![m40](https://i.imgur.com/bWXH3FVl.jpeg)
+
+*This is very simple 5x5 macropad for general use. It does not come with any particular layout, it is assumed that layouts will be built on the qmk configurator site. Row and column pins start on the edge of Arduino, if smaller board is required, you can simply connect only first x row/column pairs to the board and get the size of the board you need.*
+
+* Keyboard Maintainer: [Tomek](https://github.com/m40)
+* Hardware Supported: *3D Printed, handwired, using Arduino Pro-Micro (Atmega32U4) for controller*
+* Hardware Availability: *https://www.thingiverse.com/thing:5239739*
```
Comment (fauxpark) [inline keyboards/handwired/m40/5x5_macropad/readme.md:9]:
```suggestion
This is very simple 5x5 macropad for general use. It does not come with any particular layout, it is assumed that layouts will be built on the qmk configurator site. Row and column pins start on the edge of Arduino, if smaller board is required, you can simply connect only first x row/column pairs to the board and get the size of the board you need.

* Keyboard Maintainer: [Tomek](https://github.com/m40)
* Hardware Supported: 3D Printed, handwired, using Arduino Pro-Micro (Atmega32U4) for controller
* Hardware Availability: https://www.thingiverse.com/thing:5239739
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — COMMENTED** [review]
Is @m40 an alternate account of yours?

**Comment by m40-dev** [comment]
> Is @M40 an alternate account of yours?

@noroadsleft - no, m40 is my typical username (also on discord) but on github it was already taken.

File: keyboards/handwired/m40/5x5_macropad/5x5_macropad.c [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 Tomek (@m40)
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/5x5_macropad.c:1]:
```suggestion
// Copyright 2022 Tomek (@m40-dev)
```

File: keyboards/handwired/m40/5x5_macropad/5x5_macropad.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 Tomek (@m40)
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/5x5_macropad.h:1]:
```suggestion
// Copyright 2022 Tomek (@m40-dev)
```

File: keyboards/handwired/m40/5x5_macropad/5x5_macropad.h [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/5x5_macropad.h:16]:
```suggestion
#define LAYOUT_ortho_5x5( \
```

File: keyboards/handwired/m40/5x5_macropad/config.h [hunk 7]
```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Tomek (@m40)
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/config.h:1]:
```suggestion
// Copyright 2022 Tomek (@m40-dev)
```

File: keyboards/handwired/m40/5x5_macropad/config.h [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+/* Bootmagic Lite key configuration */
+//#define BOOTMAGIC_LITE_ROW 0
+//#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/config.h:96]: `NO_ACTION_MACRO` and `NO_ACTION_FUNCTION` no longer do anything.

`BOOTMAGIC_LITE_ROW` and `BOOTMAGIC_LITE_COLUMN` only need to be defined if you're using something other than Row 0, Column 0 - this is the default setting in QMK Firmware.

```suggestion
```

File: keyboards/handwired/m40/5x5_macropad/info.json [hunk 9]
```diff
@@ -0,0 +1,41 @@
+{
+    "keyboard_name": "5x5_macropad", 
+    "url": "https://www.thingiverse.com/thing:5239739", 
+    "maintainer": "m40",
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/info.json:4]:
```suggestion
    "maintainer": "m40-dev", 
```

File: keyboards/handwired/m40/5x5_macropad/info.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
+{
+    "keyboard_name": "5x5_macropad", 
+    "url": "https://www.thingiverse.com/thing:5239739", 
+    "maintainer": "m40", 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/info.json:6]:
```suggestion
        "LAYOUT_ortho_5x5": {
```

File: keyboards/handwired/m40/5x5_macropad/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2022 Tomek (@m40)
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/keymaps/default/keymap.c:1]:
```suggestion
// Copyright 2022 Tomek (@m40-dev)
```

File: keyboards/handwired/m40/5x5_macropad/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+	_BASE,
+	_FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/keymaps/default/keymap.c:14]:
```suggestion
    [_BASE] = LAYOUT_ortho_5x5(
```

File: keyboards/handwired/m40/5x5_macropad/keymaps/default/readme.md [hunk 13]
```diff
@@ -0,0 +1 @@
+# The default keymap for m40
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/keymaps/default/readme.md:1]:
```suggestion
# The default keymap for 5x5_macropad
```

File: keyboards/handwired/m40/5x5_macropad/readme.md [hunk 14]
```diff
@@ -0,0 +1,27 @@
+# m40
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/readme.md:1]:
```suggestion
# 5x5_macropad
```

File: keyboards/handwired/m40/5x5_macropad/readme.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+    make handwired/m40/5x5_macropad:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/readme.md:25]: You have Bootmagic Lite disabled, so the first method won't work.

```suggestion
Enter the bootloader in 2 ways:

```

File: keyboards/handwired/m40/5x5_macropad/readme.md [hunk 16]
```diff
@@ -0,0 +1,27 @@
+# m40
+
+![m40](https://i.imgur.com/bWXH3FVl.jpeg)
+
+This is very simple 5x5 macropad for general use. It does not come with any particular layout, it is assumed that layouts will be built on the qmk configurator site. Row and column pins start on the edge of Arduino, if smaller board is required, you can simply connect only first x row/column pairs to the board and get the size of the board you need.
+
+* Keyboard Maintainer: [Tomek](https://github.com/m40-dev)
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/readme.md:7]: If you wanted to put your Discord username here, that would be okay.

**Review by noroadsleft — CHANGES_REQUESTED** [review]
* changes the username references that are intended to reference the GitHub username to `m40-dev`
* renames `LAYOUT` to `LAYOUT_ortho_5x5`

GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/16288/files).

File: keyboards/handwired/m40/5x5_macropad/readme.md [hunk 17]
```diff
@@ -1,10 +1,10 @@
-# m40
+# 5x5_macropad
 
 ![m40](https://i.imgur.com/bWXH3FVl.jpeg)
 
 This is very simple 5x5 macropad for general use. It does not come with any particular layout, it is assumed that layouts will be built on the qmk configurator site. Row and column pins start on the edge of Arduino, if smaller board is required, you can simply connect only first x row/column pairs to the board and get the size of the board you need.
 
-* Keyboard Maintainer: [Tomek](https://github.com/m40-dev)
+* Keyboard Maintainer: [Tomek](https://github.com/m40-dev, discord: m40#4792)
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/readme.md:7]:
```suggestion
* Keyboard Maintainer: [Tomek](https://github.com/m40-dev) (discord: m40#4792)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Also one change request from the last review is still pending:

* https://github.com/qmk/qmk_firmware/pull/16288/files/878a27a2522e7dd664a14700a480c553849d9aaa..9895c8d90fc9eadc4be0e5307e2c6e74b21d8079#r827571793

**Review by m40-dev — COMMENTED** [review]
All files up-to-date

File: keyboards/handwired/m40/5x5_macropad/info.json [hunk 18]
```diff
@@ -0,0 +1,41 @@
+{
+    "keyboard_name": "5x5_macropad", 
+    "url": "https://www.thingiverse.com/thing:5239739", 
+    "maintainer": "m40-dev", 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/handwired/m40/5x5_macropad/info.json:6]: QMK Configurator won't render the board layout properly unless this is changed.

```suggestion
        "LAYOUT_ortho_5x5": {
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**